### ISSUE TITLE
False positive on no-member for inferred classes that are ignored

### ISSUE THREAD

**Issue by pcorpet** [PyCQA/pylint#1276]
### Steps to reproduce
1. get some code that end up having an instance that can be one of multiple classes, at least one of them being ignored for typecheck (using `ignored-modules` or `ignored-classes`), then use a non existent member on this instance. E.g.[`schroedinger.py` example file](https://gist.github.com/pcorpet/a0eaa282754a95273bccacd07419d1f7)
2. run the commandline with pylint and make sure you ignore one of the classes (here I've added extra commandline flags to make sure nothing interferes with the result) 
```sh
pylint --rcfile=/dev/null --ignored-classes=MyVeryOwnError --reports=no --disable=R schroedinger.py
```

### Current behavior
A `no-member` error is shown for the culprit line and mentions only the inferred classes that are not ignored.
```
E: 26,14: Instance of 'ValueError' has no 'node' member (no-member)
```

### Expected behavior
No error at all.

### pylint --version output
```
pylint 1.6.4, 
astroid 1.4.8
Python 3.5.2 (default, Nov 17 2016, 22:42:56) 
[GCC 4.9.2]
```

### My Guess

I believe that what happens is that the [continue](https://github.com/PyCQA/pylint/blob/69f480dfe698b17756f97ca61cd7c6390e3e7f66/pylint/checkers/typecheck.py#L627) statement which covers for ignored owners should validate the member instead of trusting other inferred owers to know better. So I would replace it by a `break`. Any thoughts?

**Comment by PCManticore** [comment]
This is an example of the control flow support we are currently lacking. pylint can't know what branch will be taken and what value ``owner`` will have, which is why it will be using all the branches of the ``if`` statement for gathering the type of ``owner``.

**Comment by pcorpet** [comment]
@PCManticore thanks for answering. What is troubling is that PyLint does not return any error if the class is not ignored. I understand that PyLint cannot know what branch is taken and thus uses all the alternatives for gathering the type of `owner`. But then if that's the case, if one of the possible types of `owner` is ignored for `no-member` errors then I believe that it should ignore those errors instead of falling back to other types.
